portfolio/src/main/java/com/google/sps/servlets/ChartServlet.java

```diff
@@ -0,0 +1,141 @@
+package com.google.sps.servlets;
+
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
+import com.google.appengine.api.datastore.EntityNotFoundException;
+import com.google.gson.Gson;
+import java.util.Calendar;
+import java.util.Date;
+import java.util.List;
+import java.util.ArrayList;
+import java.time.DayOfWeek;
+import java.time.LocalDate; 
+
+@WebServlet("/charts")
+public class ChartServlet extends HttpServlet {
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    response.setContentType("text/html;");
+
+    //Get the day of the week
+    LocalDate localdate = LocalDate.now();
+    DayOfWeek dotw = DayOfWeek.from(localdate);
+    String currentDay = (String)dotw.name(); 
+    response.getWriter().println((String)dotw.name());
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    Query query = new Query("dayOfWeek");
+    PreparedQuery results = datastore.prepare(query);
+
+    //Gets stored date from datastore
+    String storedDate ="";
+    for(Entity entity:results.asIterable()){
+        storedDate = (String)entity.getProperty("dayOfWeek");
```

Comment:
Are you trying to get the most recent date here? Cause if so wouldn't you not need to consider every entry?

Reply:
dayOfWeek only holds one value so the for-loop is definitely overdoing it but it's the only way I could figure out how to get it from the datastore. If you any insight on just grabbing one entity from the datastore, that would be much appreciated.

Reply:
Why can't dayOfWeek just be calculated from System.currentTimeMillis()?  I don't follow why we need to store a single global piece of data in datastore.